site/src/pages/StarterTemplatePage/StarterTemplatePage.tsx

```diff
@@ -1,31 +1,30 @@
-import { useMachine } from "@xstate/react";
 import { useOrganizationId } from "hooks/useOrganizationId";
 import { FC } from "react";
 import { Helmet } from "react-helmet-async";
 import { useParams } from "react-router-dom";
 import { pageTitle } from "utils/page";
-import { starterTemplateMachine } from "xServices/starterTemplates/starterTemplateXService";
 import { StarterTemplatePageView } from "./StarterTemplatePageView";
+import { useQuery } from "@tanstack/react-query";
+import { templateExamples } from "api/queries/templates";
 
 const StarterTemplatePage: FC = () => {
   const { exampleId } = useParams() as { exampleId: string };
   const organizationId = useOrganizationId();
-  const [state] = useMachine(starterTemplateMachine, {
-    context: {
-      organizationId,
-      exampleId,
-    },
-  });
+  const templateExamplesQuery = useQuery(templateExamples(organizationId));
```

Comment:
Not something to do in this PR, but just a note that maybe we need a way to get a template by ID, seems unfortunate to query all the templates just to get one!

Reply:
I do agree with you!